src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -600,8 +600,24 @@ void Creature::Update(uint32 diff)
                 m_regenTimer -= diff;
                 if (m_regenTimer <= 0)
                 {
-                    if (!IsInEvadeMode() && (!IsInCombat() || IsPolymorphed())) // regenerate health if not in combat or if polymorphed
-                        RegenerateHealth();
+                    if (!IsInEvadeMode())
+                    {
+                        // regenerate health if not in combat or if polymorphed)
+                        if (!IsInCombat() || IsPolymorphed())
+                            RegenerateHealth();
```

Comment:
If you're trying to mimic the official behavior we can see in the video I've linked before, it was not a simple "healt regen", it was an actual evade.

Reply:
That's not on the first pull, it's whenever player is not in reach after they already had engaged.

Reply:
Ok probably it's what happens right after (?)